### PR TITLE
Delete the friend

### PR DESCRIPTION
**Pull request by qkreltms**
## Description
Implement deleting the friend by it's id when user hits the friend and hits delete button on ```profile modal``` in ```Friend``` screen.
ezgif-2-aaf0e340920f

## Checklist

- [X] I read the [Contributor Guide](https://github.com/dooboolab/hackatalk-mobile/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] Run `yarn lint && yarn tsc`
- [X] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [X] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: src/components/shared/__tests__/ProfileModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    });
+
+    return createTestElement(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <Friend />
+      </MockedProvider>);
+  };
+
+  it('Render without crashing', () => {
+    const component = render(setup());
```
Comment (geoseong) [inline src/components/shared/__tests__/ProfileModal.test.tsx]: I ran `yarn test` this file, throw error. so I tried to add this below, error disappeared.
so consider put this code.
```suggestion
  await wait(() => {
    const component = render(setup());
    expect(component).toMatchSnapshot();
  });
```

File: src/components/shared/ProfileModal.tsx [hunk 2]
```diff
@@ -109,6 +111,9 @@ const styles: Styles = {
 
 const Shared = forwardRef<Ref, Props>((props, ref) => {
   let modal: Modal | null;
+  const [deleteFriendMutation, {
+    error: deleteFriendError, loading: deleteFriendLoading,
+  }] = useMutation<{ id: string }, MutationDeleteFriend>(MUTATION_DELETE_FRIEND);
```
Comment (geoseong) [inline src/components/shared/ProfileModal.tsx]: I found the way how to refetch and get friend data without using `useFriendContext`.
It seems to work well, so just put this code like @godon019 's [addFriendMutation's refetchQueries style](https://github.com/geoseong/hackatalk-mobile/blob/feat%2Fusermgmt/src/components/shared/ProfileModal.tsx#L127).

please refer to [apollo's document](https://www.apollographql.com/docs/react/api/react-hooks/#options-2) how to use `refetchQueries` option.

```suggestion
  }] = useMutation<{ id: string }, MutationDeleteFriend>(MUTATION_DELETE_FRIEND, {
    refetchQueries: () => [{ query: QUERY_FRIENDS }],
  });
```

File: src/components/shared/__tests__/ProfileModal.test.tsx (hunk 1 above)
Comment (qkreltms) [reply src/components/shared/__tests__/ProfileModal.test.tsx]: I don't know why an error occurs when I didn't warp await wait(() => { ... }) do you know why is it?
```
  console.error node_modules/react-native/Libraries/YellowBox/YellowBox.js:63
    Warning: An update to Screen inside a test was not wrapped in act(...).
```

Comment (geoseong) [reply src/components/shared/__tests__/ProfileModal.test.tsx]: I've have curious about this situation, and I googled then found a relative reference.
- [github issue: this is just a warning](https://github.com/testing-library/native-testing-library/issues/91#issuecomment-564625613)

so only way to remove this warning, we have to find something to configure.
I think many people are suffering from this situation. here is [other github issues](https://github.com/testing-library/native-testing-library/issues/25)

File: src/components/shared/__tests__/ProfileModal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          },
+        },
+      },
+    ];
+
+    const component = render(setup(mocks));
+    const friend = component.getByTestId('user-id-0');
+    fireEvent.press(friend);
+
+    // TODO: find profileModal's delete button and press it.
```
[the remark this answers is not in the document]
Comment (geoseong) [reply src/components/shared/__tests__/ProfileModal.test.tsx]: and now I see this codes,  you tried to render `Friend` component.
I understand what you were trying to do but I think you try to render `ProfileModal` component in this test code.

here is a code I've tried to.
```suggestion
import 'react-native';

import * as React from 'react';

import {
  RenderResult,
  render,
  wait,
} from '@testing-library/react-native';
import { createTestElement, createTestProps } from '../../../../test/testUtils';

import Shared from '../ProfileModal';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

describe('[ProfileModal] rendering test', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Shared {...props} />);
    testingLib = render(component);
  });

  it('renders without crashing', async () => {
    const { baseElement } = testingLib;
    await wait(() => {
      expect(baseElement).toMatchSnapshot();
      expect(baseElement).toBeTruthy();
    });
  });
});

```

**Comment by geoseong** [comment]
@qkreltms 
PR merged!

don't forget to update snapshot via `yarn test -u` when you change test code next time.

**Pull request merged**